src/common/util.cpp

```diff
@@ -195,6 +195,57 @@ namespace tools
     catch (...) {}
   }
 
+  file_locker::file_locker(const std::string& filename)
+  {
+#ifdef WIN32
+    m_fd = INVALID_HANDLE_VALUE;
+    WCHAR wide_path[1000];
```

Comment:
Is there no PATH_MAX for Windows ?

Reply:
This was copied from code added in #3313. It seems like Windows also has _PATH_MAX, but the value might change depending on different versions of Windows, and it doesn't seem like the universal way of guaranteeing sufficient length for path names. The current codebase has two places where `MultiByteToWideChar` is used:

https://github.com/monero-project/monero/blob/ffab67004ca9a846d6f6d39de552e7ce79febf88/external/db_drivers/liblmdb/mdb.c#L10942-L10959

https://github.com/monero-project/monero/blob/ffab67004ca9a846d6f6d39de552e7ce79febf88/tests/gtest/src/gtest.cc#L858-L869

In both cases, the length of the resulting UTF-16 string is obtained by first calling the function with the output location set to NULL, which seems like the right way.

If this way is to be used, I think it makes sense to also replace the corresponding code in file_io_utils. What do you think? CC: @rbrunner7 @hyc

Reply:
Guilty ... I knew that, but when I wrote that code I was on my last legs after working so long on that UTF-8 filename stuff that I simply hacked it in the simplest possible way.

I agree that it should be replaced sooner or later.

Reply:
Agreed, that's the safe way to do it.